plugins/woocommerce-admin/client/marketplace/components/search/search.tsx

```diff
@@ -64,13 +67,19 @@ function Search(): JSX.Element {
 		}
 	};
 
+	const onClose = () => {
+		setSearchTerm( '' );
+		runSearch( '' );
+	};
+
 	return (
 		<SearchControl
 			label={ searchPlaceholder }
 			placeholder={ searchPlaceholder }
 			value={ searchTerm }
 			onChange={ setSearchTerm }
 			onKeyUp={ handleKeyUp }
+			onClose={ onClose }
```

Comment:
Wow, there's a `onClose`? That is neat!

Reply:
Yup, it's in the docs https://wordpress.github.io/gutenberg/?path=/docs/components-searchcontrol--docs :)